cartographer_grpc/framework/rpc.h

```diff
@@ -33,33 +37,45 @@ class Rpc {
   enum class State { NEW_CONNECTION = 0, READ, WRITE, DONE };
   struct RpcState {
     const State state;
+    Service* service;
     Rpc* rpc;
   };
 
-  Rpc(const RpcHandlerInfo& rpc_handler_info);
+  Rpc(int method_index, ::grpc::ServerCompletionQueue* server_completion_queue,
+      const RpcHandlerInfo& rpc_handler_info, Service* service);
 
+  int method_index() const { return method_index_; }
+  ::grpc::ServerCompletionQueue* server_completion_queue();
   ::grpc::internal::RpcMethod::RpcType rpc_type() const;
   ::grpc::ServerContext* server_context() { return &server_context_; }
-  ::grpc::internal::ServerAsyncStreamingInterface* responder();
-  RpcState* GetState(State state);
+  ::grpc::internal::ServerAsyncStreamingInterface* streaming_interface();
+  RpcState* GetRpcState(State state);
```

Comment:
Shouldn't this be a static function that depends on rpc*, service*, and State and then we can drop the state_ members and the Service* from the constructor arguments.

Reply:
We use the memory address as a void* tag for the gRPC requests. So we need to allocate for each RPC object all states.